pandas/tests/indexes/common.py

```diff
@@ -721,3 +736,19 @@ def test_fillna(self):
                 expected[1] = True
                 self.assert_numpy_array_equal(idx._isnan, expected)
                 self.assertTrue(idx.hasnans)
+
+    def test_copy(self):
+        # GH12309
+        for name, index in compat.iteritems(self.indices):
+            first = index.__class__(index, copy=True, name='mario')
```

Comment:
why do you have first and second here? you are not testing second till below. this is not clear at all what you are actually testing.